test/allow-extended-operators.test.js

```diff
@@ -51,12 +36,193 @@ describe('Model.settings.allowExtendedOperators', () => {
     constructor(dataSource) {
     }
 
+    create(model, data, options, callback) {
+      callback();
+    }
+
+    updateAttributes(model, id, data, options, callback) {
+      callback();
+    }
+
     all(model, filter, options, callback) {
       // return the raw "value" query
-      var instanceFound = {
+      let instanceFound = {
         value: filter.where.value,
       };
       callback(null, [instanceFound]);
     }
   }
+
+  describe('connector.settings.allowExtendedOperators', () => {
+    context('DAO.find()', () => {
+      it('converts extended operators to string value by default', () => {
+        const TestModel = createTestModel();
+        return TestModel.find(extendedQuery()).then((results) => {
+          should(results[0].value).eql('[object Object]');
+        });
+      });
+
+      it('preserves extended operators with allowExtendedOperators set', () => {
+        const TestModel = createTestModel({allowExtendedOperators: true});
+        return TestModel.find(extendedQuery()).then((results) => {
+          should(results[0].value).eql({$exists: true});
+        });
+      });
+
+      function extendedQuery() {
+        // datasource modifies the query,
+        // we have to build a new object for each test
+        return {where: {value: {$exists: true}}};
+      }
+    });
+
+    context('DAO.updateAttributes()', () => {
+      it('`options.allowExtendedOperators` override connector settings - disable strict check', () => {
+        const TestModel = createTestModel({allowExtendedOperators: false}, {strict: true});
+        return TestModel.create({value: 'test'}).then(function(instance) {
+          let updateData = {$set: {value: 'changed'}};
+          return instance.updateAttributes(updateData, {allowExtendedOperators: true})
+            .then(function() {
+              should(TestModel.lastPersistedData).eql(updateData);
+            });
+        });
+      });
+
+      it('`options.allowExtendedOperators` override connector settings - enable strict check', () => {
+        const TestModel = createTestModel({allowExtendedOperators: true}, {strict: true});
+        return TestModel.create({value: 'test'}).then(function(inst) {
+          return inst.updateAttributes({$set: {value: 'changed'}}, {allowExtendedOperators: false})
+            .catch(function(err) {
```

Comment:
You need to handle the case when `updateAttributes` did not throw any error.

I also think it's safe to assume `err` will be defined when a promise was rejected. To be safe, your assertion can be rewritten in a way that supports undefined `err` too.

```js
return inst.updateAttributes(...).then(
  () => throw new Error('updateAttributes() should have failed.'),
  (err) => should(err).have.property('name', 'ValidationError'));
```